### ISSUE TITLE
We should not start processing gossips until initialization is finished

### ISSUE THREAD

**Issue by ghost-not-in-the-shell** [MinaProtocol/mina#3543]
In current implementation we would start listening and processing gossips from the network as soon as we start the daemon. And there's a problem with that in integration test, because with eager bootstrap, there would be an initialization phase which blocks bootstrap controller from consuming the transitions from gossip. This would cause the transitions being stacked up in `bootstrap_pipe` in transition_router and possibly leads to an overflow in that pipe.

There's 2 possible solution to that:
1. Do not start listening to gossips until initialization is finished or failed.
2. Do not start processing gossips related to transitions until initialization is finished or failed.

The first option is more conservative because I don't think we should not start processing gossips related to transactions until we have a working transition frontier because of the nonce issue. For this reason, I even want hold off transaction-related gossip processing until catchup finishes. And it's less meaningful to processing gossips related to transitions because we would gather best tip from peers during initialization phase. From bootstrap's point of view, best tip from the peers is the only thing we need to make bootstrap controller start working.

**Comment by enolan** [comment]
We already reject all gossipped incoming transactions if the frontier isn't
available. When it is available we validate them based on the current best
tip, which may be out of date during catchup. If we turned off processing
during catchup we'd lose potentially valuable transactions sometimes.

**Comment by bkase** [comment]
@ghost-not-in-the-shell will talk with @enolan and produce a state transition diagram to explain initialization -- this should fix secular grail

**Comment by bkase** [comment]
put the state transition diagram on discourse (the forums)

**Comment by ghost-not-in-the-shell** [comment]
I already have a hack that mitigates the bootstrap overflow in integration tests: #3545. But I still believe the right thing to do is that postpone the procession of gossip until initialization finishes. After discuss with @enolan, I realized that the `transaction_pool_diff` and `snark_pool_diff`, those gossips would be discarded until we have a working frontier. I believe we should do the same for `new_state` gossip too (discard those gossip until initialization finishes which is exactly what I did in #3545).

This PR #3653 contains a diagram I draw that reflects the control flow of transition router.
